src/registry/map/validationSchema.ts

```diff
@@ -0,0 +1,64 @@
+import type {MapComponentSchema} from '@open-formulieren/types';
+import {z} from 'zod';
+
+import type {GetValidationSchema} from '@/registry/types';
+
+const getValidationSchema: GetValidationSchema<MapComponentSchema> = (
+  componentDefinition,
+  {validatePlugins}
+) => {
+  const {key, validate = {}, errors} = componentDefinition;
+  const {required, plugins = []} = validate;
+
+  const coordinates = z.array(z.number()).length(2);
+  const pointValue = coordinates;
+  const lineValue = z.array(coordinates).min(1);
+  const polygonValue = z.array(z.array(coordinates).min(1)).length(1);
+  let schema: z.ZodFirstPartySchemaTypes = z.union([
+    z.object({
+      type: z.literal('Point'),
+      coordinates: pointValue,
+    }),
+    z.object({
+      type: z.literal('LineString'),
+      coordinates: lineValue,
+    }),
+    z.object({
+      type: z.literal('Polygon'),
+      coordinates: polygonValue,
+    }),
+    z.null(),
+  ]);
```

Comment:
this can be made more strict, and then should also make the value parsing error handling obsolete in https://github.com/open-formulieren/formio-renderer/pull/214/files#diff-a8399940fa4c9f7a4d69aaf85be8dce8c4455d5ddc686e024603223e64493a3dR81 - which I suspect was only added to be able to introduce an invalid state with non-sense data.

You can inspect `componentDefinition.interactions` to check which shapes are allowed and compose the schema that way. Also, looking at `validate.required` determines whether null is allowed or not.

I'm adding a commit to implement gradual schema building, you can then set up a story that dissallows a particular type but provide valid geojson to trigger the error :)